activerecord/lib/active_record/persistence.rb

```diff
@@ -55,6 +57,201 @@ def create!(attributes = nil, &block)
+        insert_all([ attributes ], **options)
+      end
+
+      # Inserts multiple records into the database. This method constructs a single SQL INSERT
+      # statement and sends it straight to the database. It does not instantiate the involved
+      # models and it does not trigger Active Record callbacks or validations. However, values
+      # passed to #insert_all will still go through Active Record's normal type casting and
+      # serialization.
+      #
+      # The +attributes+ parameter is an Array of Hashes. These Hashes describe the
+      # attributes on the objects that are to be created. All of the Hashes must have
+      # same keys.
+      #
+      # Records that would violate a unique constraint on the table are skipped.
+      #
+      # Returns an `ActiveRecord::Result`. The contents of the result depend on the
+      # value of `:returning` (see below).
+      #
+      # ==== Options
+      #
+      # [:returning]
+      #   (Postgres-only) An array of attributes that should be returned for all successfully
+      #   inserted records. For databases that support `INSERT ... RETURNING`, this will default
+      #   to returning the primary keys of the successfully inserted records. Pass
+      #   `returning: %w[ id name ]` to return the id and name of every successfully inserted
+      #   record or pass `returning: false` to omit the clause.
+      #
+      # [:unique_by]
+      #   (Postgres and SQLite only) In a table with more than one unique constaint or index,
+      #   new records may considered duplicates according to different criteria. By default,
+      #   new rows will be skipped if they violate _any_ unique constraint/index. By defining
+      #   :unique_by, you can skip rows that would create duplicates according to the given
+      #   constraint but raise `ActiveRecord::RecordNotUnique` if rows violate other constraints.
+      #
+      #   (For example, maybe you assume a client will try to import the same ISBNs more than
+      #   once and want to silently ignore the duplicate records, but you don't except any of
+      #   your code to attempt to create two rows with the same primary key and would appreciate
+      #   an exception report in that scenario.)
+      #
+      #   Indexes can be identified by an array of columns:
+      #
+      #     unique_by: { columns: %w[ isbn ] }
+      #
+      #   Partial indexes can be identified by an array of columns and a <tt>:where</tt> condition:
+      #
+      #     unique_by: { columns: %w[ isbn ], where: "published_on IS NOT NULL" }
+      #
+      # ==== Example
+      #
+      #   # Insert multiple records and skip duplicates
+      #   # ('Eloquent Ruby' will be skipped because its id is duplicate)
+      #   Book.insert_all([
+      #     { id: 1, title: 'Rework', author: 'David' },
+      #     { id: 1, title: 'Eloquent Ruby', author: 'Russ' }
+      #   ])
+      #
+      def insert_all(attributes, **options)
```

Comment:
Use kwargs instead of **options to make it clear what the options are. There are only two!